File: PROJECT_C++/PROJECT_C++/Trawa.cpp
#include "Trawa.h"
#include <iostream>
#include "Swiat.h"
#include <string>
#include <vector>

Trawa::Trawa(int x, int y, Swiat& swiat) : Roslina(0, x, y, swiat) {};

void Trawa::rysowanie() {
	std::cout << "*";
}

void Trawa::rozmnozSie(int x, int y, Swiat& swiat) {
	Trawa* nowyOrganizm = new Trawa(x, y, swiat);
	swiat.dodajOrganizm(*nowyOrganizm);
}

Trawa::~Trawa() {};

File: OOPJava/src/main/game/rosliny/Trawa.java
package main.game.rosliny;

import main.game.*;
public class Trawa extends Roslina{
    public Trawa(int x, int y, Swiat swiat){ super(0,x,y,swiat); }
    public Trawa(int sila, int inicjatywa, int x, int y, int poprzednieX, int poprzednieY, int wiek, Swiat swiat) {
        super(sila, inicjatywa, x, y, poprzednieX, poprzednieY, wiek, swiat);
    };

    @Override
    public String rysowanie(){
        return "trawa.png";
    }

    @Override
    public void rozmnozSie(int x, int y, Swiat swiat){
        Trawa nowyOrganizm = new Trawa(x, y, swiat);
        swiat.dodajOrganizm(nowyOrganizm);
    }

    @Override
    public String getImie() { return "Trawa"; };
}
